lib/sof/cycle/parser.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/hash/keys"
+require "active_support/core_ext/object/blank"
+require "active_support/core_ext/object/inclusion"
+require "active_support/core_ext/hash/reverse_merge"
+require "active_support/isolated_execution_state"
+
+module SOF
+  # This class is not intended to be referenced directly.
+  # This is an internal implementation of Cycle behavior.
+  class Cycle
+    class Parser
```

Comment:
:art: Would it be possible to make this a module instead of a class in order to avoid the [Nested Classes](https://alchemists.io/articles/ruby_antipatterns#_nested_classes) antipattern?

Reply:
That would be a good thing, yes. In future changes perhaps